### DISCUSSION TITLE
Issue with update subscription

### DISCUSSION THREAD

**Discussion by jychua95** [impress-org/givewp#7245] [Developer Q&A]
Hi, 

I attempted to adjust some subscription details and update them using the `save();` method in Subscription.php, but it didn't work as expected. The details weren't updated as intended.

As example: 
```
$subscription_id = '20';
$subscription = Subscription::find($subscription_id);
//update transactionID
$subscription->transactionId = 'abcde112345';
               
//save the subscription
$subscription->save(); <-- this not working,
```

I would greatly appreciate any guidance or assistance on this matter. Thank you!

**Comment by kjohnson** [comment]
Are you seeing any `Exception`s being thrown?

When you say it isn't updating as intended, are you looking for changes in the Subscriptions UI in the WordPress admin, or are you checking the `transactionId` property on the `$subscription` model?

Looking at the code, I see no reason why that wouldn't be working, so if I can better understand what you are experiencing then maybe we can find a solution.

**Reply by jychua95** [reply]
Hello,

I created a custom plugin to manage custom recurring donations in collaboration with a payment gateway. The transactionID serves as a token that stores the customer's credit/debit card details on the payment gateway's side. Just for your information, the token was provided by the payment gateway side when the card was saved.

There's a scenario where customers might update their saved card information. Consequently, I needed to implement a custom function to update the transactionID whenever the saved card information is modified.

It appears that the `save()` method is not functioning as expected, and no exceptions have been thrown either.

Since my payment gateway lacks support for the recurring feature, I have no alternative but to proceed in this manner. 

If there's another alternative for saving the token, I'm open to exploring it. Currently, I'm using the transactionID, but I acknowledge that this might not be the best practice.
